src/init.cpp

```diff
@@ -900,6 +905,49 @@ bool AppInit2(Config &config, boost::thread_group &threadGroup, CScheduler &sche
     LOGA("Using at most %i connections (%i file descriptors available)\n", nMaxConnections, nFD);
     std::ostringstream strErrors;
 
+    // bip135 begin
+    // check for fork deployment CSV file, read it
+    string ForksCsvFile = GetForksCsvFile().string();
+
+    if (boost::filesystem::exists(ForksCsvFile))
+    {
+        ifstream csvFile;
+        bool CsvReadOk = true;
+        try
+        {
+            csvFile.open(ForksCsvFile.c_str(), ios::in);
+            if (csvFile.fail())
+            {
+                throw std::runtime_error("unable to open deployment file for reading");
+            }
+
+            LOGA("Reading deployment configuration CSV file at '%s'\n", ForksCsvFile);
+            // read the CSV file and apply the parameters for current network
+            CsvReadOk = ReadForksCsv(chainparams.NetworkIDString(), csvFile, chainparams.GetModifiableConsensus());
+            csvFile.close();
+        }
+        catch (const std::exception &e)
+        {
+            LOGA("Unable to read '%s'\n", ForksCsvFile);
+            // if unable to read file which is present: abort
+            return InitError(strprintf(
+                _("Warning: Could not open deployment configuration CSV file '%s' for reading"), ForksCsvFile));
+        }
+        // if the deployments data doesn't validate correctly, shut down for safety reasons.
+        if (!CsvReadOk)
+        {
+            LOGA("Validation of '%s' failed\n", ForksCsvFile);
+            return InitError(strprintf(
+                _("Deployment configuration file '%s' contained invalid data - see debug.log"), ForksCsvFile));
+        }
+    }
+    else
+    {
+        // be noisy, but don't fail if file is absent - use built-in defaults
+        LOGA("No deployment configuration found at '%s' - using defaults\n", ForksCsvFile);
```

Comment:
If the user configured a fork file but we can't find it, I think that we should exit with an error.  

If the default fork file is not there, its ok to log and continue.